Formula/rustfmt.rb

```diff
@@ -20,17 +20,50 @@ class Rustfmt < Formula
 
   depends_on "rustup-init" => :build
   depends_on "rust" => :test
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "patchelf" => :build
+  end
 
   def install
-    system "#{Formula["rustup-init"].bin}/rustup-init", "-qy", "--no-modify-path"
-    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
     # we are using nightly because rustfmt requires nightly in order to build from source
-    # pinning to nightly-2021-11-08 to avoid inconstency
-    nightly_version = "nightly-2021-11-08"
-    components = %w[rust-src rustc-dev llvm-tools-preview]
-    system "rustup", "toolchain", "install", nightly_version
+    # pinning to nightly-2023-07-01 to avoid inconstency
+    nightly_version = "nightly-2023-07-01"
+    rust_up_args = [
+      "--profile", "minimal", "-y", "--no-modify-path",
+      "--default-toolchain", nightly_version
+    ]
+    system "#{Formula["rustup-init"].bin}/rustup-init", *rust_up_args
+    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
+    components = %w[rustc-dev llvm-tools]
     system "rustup", "component", "add", *components, "--toolchain", nightly_version
     system "cargo", "install", *std_cargo_args
```

Comment:
I know this was already here, but `rustup`'s `cargo` respects the `rust-toolchain` file so we don't need to duplicate it:
```suggestion
    system "#{Formula["rustup-init"].bin}/rustup-init", "-qy", "--no-modify-path"
    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
    system "cargo", "install", *std_cargo_args
```

Reply:
Thanks for the advice.